modules/document_repository/ajax/documentEditUpload.php

```diff
@@ -52,34 +44,39 @@
     || $userSingleton->hasPermission('document_repository_delete')
 ) {
     if ($action == 'upload') {
-        $puser      = $_POST['user'];
-        $category   = $_POST['category'];
-        $site       = $_POST['site']       !== '' ? $_POST['site'] : null;
-        $instrument = $_POST['instrument'] !== '' ? $_POST['instrument'] : null;
-        $pscid      = $_POST['pscid']      !== '' ? $_POST['pscid'] : null;
-        $visit      = $_POST['visit']      !== '' ? $_POST['visit'] : null;
-        $comments   = $_POST['comments']   !== '' ? $_POST['commnets'] : null;
-        $version    = $_POST['version']    !== '' ? $_POST['version'] : null;
-
-        $fileSize = $_FILES["file"]["size"];
-        $fileName = $_FILES["file"]["name"];
-        $fileType = end((explode(".", $fileName)));
-
-        // __DIR__ is the document_repository ajax directory
-        // when this script is executing. Go up a level to the
-        // document_repository module directory, and use a
-        // user_uploads directory as a base for user uploads
-        $base_path = __DIR__ . "/../user_uploads/";
-        $fileBase  = $puser . "/" . $fileName;
-
-        if (!file_exists($base_path . $puser)) {
-            mkdir($base_path . $puser, 0777);
+        $category   = $_POST['category']; // required
+        $site       = $_POST['site'] ?? ''; // required but allowed to be empty
```

Comment:
@johnsaigle the error is occurring here
`SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'For_site' at row 1`

Reply:
I think it should be `$_POST['site'] ?? null`